crates/weaver_forge/src/filter.rs

```diff
@@ -4,87 +4,31 @@
-use jaq_syn::Def;
-use std::fmt::Debug;
+use std::{collections::BTreeMap, fmt::Debug};
 
 /// A filter that can be applied to a JSON value.
 pub struct Filter {
     filter_expr: String,
-    filter: jaq_interpret::Filter,
 }
 
 impl Filter {
     /// Create a new filter from a string expression or return an error if the
     /// expression is invalid.
     /// The vars parameter is a list of variable names that can be used in the
     /// filter expression.
-    pub fn try_new(filter_expr: &str, vars: Vec<String>, defs: Vec<Def>) -> Result<Self, Error> {
-        let mut ctx = jaq_interpret::ParseCtx::new(vars);
-        ctx.insert_natives(jaq_core::core());
-        ctx.insert_defs(jaq_std::std());
-        ctx.insert_defs(defs);
-
-        let (parsed_expr, errs) = jaq_parse::parse(filter_expr, jaq_parse::main());
-
-        // If there are any errors, return them
-        if !errs.is_empty() {
-            return Err(Error::CompoundError(
-                errs.into_iter()
-                    .map(|e| Error::FilterError {
-                        filter: filter_expr.to_owned(),
-                        error: e.to_string(),
-                    })
-                    .collect(),
-            ));
-        }
-
-        let parsed_expr = parsed_expr.ok_or_else(|| Error::FilterError {
-            filter: filter_expr.to_owned(),
-            error: "No parsed expression".to_owned(),
-        })?;
-        let filter = ctx.compile(parsed_expr);
-        if !ctx.errs.is_empty() {
-            return Err(Error::CompoundError(
-                ctx.errs
-                    .iter()
-                    .map(|e| Error::FilterError {
-                        filter: filter_expr.to_owned(),
-                        error: e.0.to_string(),
-                    })
-                    .collect(),
-            ));
-        }
-
-        Ok(Self {
+    pub fn new(filter_expr: &str) -> Self {
+        Self {
             filter_expr: filter_expr.to_owned(),
-            filter,
-        })
+        }
     }
 
     /// Apply the filter to a JSON value and return the result as a JSON value.
     pub fn apply(
         &self,
         ctx: serde_json::Value,
-        jq_ctx: Vec<Val>,
+        values: &BTreeMap<String, serde_json::Value>,
     ) -> Result<serde_json::Value, Error> {
-        let inputs = RcIter::new(core::iter::empty());
-        let filter_result = self.filter.run((Ctx::new(jq_ctx, &inputs), Val::from(ctx)));
-        let mut errs = Vec::new();
-        let mut values = Vec::new();
-
-        for r in filter_result {
-            match r {
-                Ok(v) => values.push(serde_json::Value::from(v)),
-                Err(e) => errs.push(e),
-            }
-        }
-
-        if values.len() == 1 {
-            return Ok(values.pop().expect("values.len() == 1, should not happen"));
-        }
-
-        Ok(serde_json::Value::Array(values))
+        crate::jq::execute_jq(&ctx, &self.filter_expr, values)
```

Comment:
Happy to see the updated signature of the `execute_jq` function in minijinja with the BTreeMap. It’s a much better!